### PR TITLE
Initial commit of sixth JAX-101 notebook

### PR DESCRIPTION
**Pull request by jakevdp**

### CODE REVIEW

File: docs/jax-101/06-parallelism.md [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+jax.vmap4(jax.pmap(f, axis_name='i'), axis_name='j')
+```
+
+A `jax.lax.psum(..., axis_name='i')` in `f` would refer only to the pmapped axis, since they share the `axis_name`. 
+
+In general, `jax.pmap` and `jax.vmap` can be nested in any order, and with themselves (so you can have a `pmap` within another `pmap`, for instance).
+
++++ {"id": "WzQHxnHkCxej"}
+
+# Example
```
Comment (skye) [inline docs/jax-101/06-parallelism.md]: I think this should be ##, it's coming out as a top-level item in the ToC

File: docs/jax-101/06-parallelism.md [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+import matplotlib.pyplot as plt
+plt.scatter(xs, ys)
+plt.plot(xs, params.weight * xs + params.bias, c='red', label='Model Prediction')
+plt.legend()
+plt.show()
+```
+
++++ {"id": "4wFJcqbhbn81"}
+
+# Aside: hosts and devices in JAX
```
Comment (skye) [inline docs/jax-101/06-parallelism.md]: Same here, ##

File: docs/jax-101/06-parallelism.md [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+  name: python3
+---
+
++++ {"id": "tCOWitsAS1EE"}
+
+# Parallel evaluation in JAX
+
+[![Open in Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/google/jax/blob/master/docs/jax-101/06-parallelism.ipynb)
+
+*Authors: TODO*
```
Comment (skye) [inline docs/jax-101/06-parallelism.md:20]: We have authors now!

File: docs/jax-101/06-parallelism.md [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+plt.plot(xs, params.weight * xs + params.bias, c='red', label='Model Prediction')
+plt.legend()
+plt.show()
+```
+
++++ {"id": "4wFJcqbhbn81"}
+
+## Aside: hosts and devices in JAX
+
+When running on TPU, the idea of a 'host' becomes important. A host is the CPU that manages several devices. A single host can only manage so many devices (usually 8, but with some tricks can be extended to 16), so when running very large parallel programs, multiple hosts are needed, and some finesse is required to manage them.
```
Comment (skye) [inline docs/jax-101/06-parallelism.md:396]: Let's not mention the part about extending to 16 devices, you can only do that internally. I would just say "(usually 8)".

**Review by skye — APPROVED** [review]

**Comment by skye** [comment]
FYI I think some of my comments got messed up because you pushed changes while I made them. Lemme know if anything's incomprehensible.

File: docs/jax-101/06-parallelism.md [hunk 5]
```diff
@@ -0,0 +1,423 @@
...
+
++++ {"id": "tCOWitsAS1EE"}
+
+# Parallel evaluation in JAX
+
+[![Open in Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/google/jax/blob/master/docs/jax-101/06-parallelism.ipynb)
+
+*Authors: TODO*
+
+In this section we will discuss the facilities built into JAX for single-program, multiple-data (SPMD) code.
```
Comment (8bitmp3) [inline docs/jax-101/06-parallelism.md:22]: Dumb question (sorry) @jakevdp: In https://github.com/google/jax/pull/5958/files, you added SPMD to the JAX glossary and defined it as "single program multi data", whereas here it's "single-program, multiple-data". Is there more than one way of writing SPMD?

File: docs/jax-101/06-parallelism.md [hunk 6]
```diff
@@ -0,0 +1,423 @@
...
+
+At some point during the update step, we need to combine the gradients computed by each device -- otherwise, the updates performed by each device would be different. That's why we use `jax.lax.pmean` to compute the mean across the `num_devices` axis, giving us the average gradient of the batch. That average gradient is what we use to compute the update.
+
+Aside on naming: here, we use `num_devices` for the `axis_name` for didactic clarity while introducing `jax.pmap`. However, in some sense that is tautologous: any axis introduced by a pmap will represent a number of devices. Therefore, it's common to see the axis be named something semantically meaningful, like `batch`, `data` (signifying data parallelism) or `model` (signifying model parallelism).
```
Comment (8bitmp3) [inline docs/jax-101/06-parallelism.md:291]: Another dumb question, sorry: is this meant to be `A side note on naming` or `Aside from naming`. (I see the PR is `pull ready` but if it's not too late).
